eslint.config.mjs

```diff
@@ -0,0 +1,339 @@
+import { fixupConfigRules, fixupPluginRules } from "@eslint/compat";
+import typescriptEslint from "@typescript-eslint/eslint-plugin";
+import prettier from "eslint-plugin-prettier";
+import tsParser from "@typescript-eslint/parser";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import js from "@eslint/js";
+import { FlatCompat } from "@eslint/eslintrc";
+import { createRequire } from "module"
+
+// Workaround for https://github.com/anza-xyz/eslint-plugin-require-extensions/issues/18
+const require = createRequire(import.meta.url)
+const requireExtensions = require("eslint-plugin-require-extensions")
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+    allConfig: js.configs.all
+});
+
+export default [{
+    ignores: [
+        "coverage/**/*",
+        "**/output/**/*",
+        "bundles/**/*",
+        "**/bundle-*",
+        "dist/**/*",
+        "examples/**/*",
+        "test262/**/*",
+        "**/*.html",
+        "**/ava*.config.js",
+        "**/.ava*.config.js",
+        "packages/cosmic-proto/node_modules/",
+        "packages/cosmic-proto/coverage/",
+        "packages/cosmic-proto/dist/",
```

Comment:
are these not implied by the simpler globs? we want to ignore these no matter where they are:
- `coverage`
- `dist`
- `node_modules`

Reply:
These were migrated from all our our current `eslintignore` files. I do think it makes sense to have these ignored gobally though